azurerm/internal/services/frontdoor/resource_arm_front_door.go

```diff
@@ -1349,21 +1390,35 @@ func flattenArmFrontDoorRoutingRule(input *[]frontdoor.RoutingRule) []interface{
 					v := brc.(frontdoor.ForwardingConfiguration)
 
 					c["backend_pool_name"] = flattenArmFrontDoorSubResource(v.BackendPool, "BackendPools")
+					// Link to issue: https://github.com/Azure/azure-sdk-for-go/issues/6762
+					if c["backend_pool_name"] == "" {
+						c["backend_pool_name"] = flattenArmFrontDoorSubResource(v.BackendPool, "backendPools")
+					}
 					c["custom_forwarding_path"] = v.CustomForwardingPath
 					c["forwarding_protocol"] = string(v.ForwardingProtocol)
 
 					if cacheConfiguration := v.CacheConfiguration; cacheConfiguration != nil {
-						if queryParameter := cacheConfiguration.QueryParameterStripDirective; queryParameter != "" {
-							c["cache_query_parameter_strip_directive"] = string(queryParameter)
+						c["cache_enabled"] = true
+						if stripDirective := cacheConfiguration.QueryParameterStripDirective; stripDirective != "" {
+							c["cache_query_parameter_strip_directive"] = string(stripDirective)
 						} else {
 							c["cache_query_parameter_strip_directive"] = string(frontdoor.StripNone)
 						}
 
-						c["cache_use_dynamic_compression"] = false
-
 						if dynamicCompression := cacheConfiguration.DynamicCompression; dynamicCompression != "" {
-							if dynamicCompression == frontdoor.DynamicCompressionEnabledEnabled {
-								c["cache_use_dynamic_compression"] = true
+							c["cache_use_dynamic_compression"] = bool(string(dynamicCompression) == string(frontdoor.DynamicCompressionEnabledEnabled))
+						}
+					} else {
+						c["cache_enabled"] = false
+
+						//get `forwarding_configuration`
+						if o, ok := oldByName[*name]; ok {
```

Comment:
Name could be nil so we'll need to nil check it

Reply:
yep yep